### ISSUE TITLE
C++ Core Guidelines: Formatting

### ISSUE THREAD

**Issue by krepa098** [snt-arg/guidelines#4]
Suggestion:
It is common practice in modern languages (e.g. Rust, Go) to enforce coherent formatting rules via formatting utilities (e.g. clang-format, rustfmt, gofmt). C++ projects should provide a .clang-format file and all code should be formatted according to that style definition before committing. This improves readability of the code and gets rid of ugly situations where some contributors use spaces and others tabs.

**Comment by hridaybavle** [comment]
Yes I agree I do use clang-format for cpp projects. What do you suggest we have a common clang-format file in the wiki which could be used by everyone in the group?

**Comment by krepa098** [comment]
I don't mind if different projects use different styles and not everyone has the same taste. It should just be consistent. E.g. I'm using
```
---
BasedOnStyle: Chromium
BreakBeforeBraces: Allman
ColumnLimit: '120'
IndentWidth: '4'
TabWidth: '4'
UseTab: Never
...
```
I guess we can use that as a starting point resp. example.

**Comment by hridaybavle** [comment]
Okay agree I at least put a common file for all and then we see how we can keep on modifying it

**Comment by hridaybavle** [comment]
@krepa098 I have added an example .clang-format file to start with you can give your comments.
